includes/class-scd-ext-quiz-frontend.php

```diff
@@ -78,13 +80,59 @@ public function quiz_content_drip_filter( $quizzes ) {
 			if ( Sensei_Content_Drip()->access_control->is_lesson_access_blocked( $lesson_id ) ) {
 				// Change the quiz content accordingly
 				$quizzes[ $index ] = $this->get_quiz_with_updated_content( $quiz );
+				// User Should not be able to view restricted quiz Questions
+				add_filter( 'sensei_can_user_view_lesson', array( $this, 'can_user_access_quiz_for_lesson' ), 20, 2 );
 			}
 		}
 
 		return $quizzes;
 	}
 
 	/**
+	 * Don't show quiz content when part of a lesson that hasn't dripped yet.
+	 * Hooked into "sensei_can_user_view_lesson"
+	 * @since 1.0.6
+	 * @param $can_user_view_lesson
+	 * @param $lesson_id
+	 * @param $user_id
+	 * @return bool
+	 */
+	public function can_user_access_quiz_for_lesson( $can_user_view_lesson, $lesson_id, $user_id = null ) {
+		if ( false == empty( $lesson_id ) && Sensei_Content_Drip()->access_control->is_lesson_access_blocked( $lesson_id ) ) {
+			return false;
+		}
+
+		return $can_user_view_lesson;
+	}
+
+	/**
+	 * Display SCD Message if this Quiz is part of a lesson that has't dripped yet.
+	 * Hooked into "sensei_single_quiz_content_inside_before"
+	 * @since 1.0.6
+	 * @param $quiz_id
+	 */
+	public function the_user_status_message( $quiz_id ) {
+		if ( empty( $quiz_id ) ) {
+			return;
+		}
+
+		$lesson_id =  Sensei()->quiz->get_lesson_id( $quiz_id );
+		if ( empty( $lesson_id ) ) {
```

Comment:
Not sure, suspect it's not. Removing this check is OK though, the next if will be false if the lesson can't be found